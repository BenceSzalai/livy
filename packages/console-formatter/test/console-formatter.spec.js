const { ConsoleFormatter } = require('../src/console-formatter')

describe('@livy/console-formatter', () => {
  it('should format records correctly', () => {
    const consoleFormatter = new ConsoleFormatter()

    const createRecord = record.with({
      message: 'Test ConsoleFormatter',
      context: { foo: 1 },
      extra: { bar: 2 }
    })

    expect(
      consoleFormatter.format(createRecord({ level: 'debug' }))
    ).toMatchSnapshot()
    expect(
      consoleFormatter.format(createRecord({ level: 'info' }))
    ).toMatchSnapshot()
    expect(
      consoleFormatter.format(createRecord({ level: 'notice' }))
    ).toMatchSnapshot()
    expect(
      consoleFormatter.format(
        createRecord({
          level: 'warning'
        })
      )
    ).toMatchSnapshot()
    expect(
      consoleFormatter.format(
        createRecord({
          recordWithContextAndExtra: createRecord,
          level: 'error'
        })
      )
    ).toMatchSnapshot()
    expect(
      consoleFormatter.format(
        createRecord({
          level: 'critical'
        })
      )
    ).toMatchSnapshot()
    expect(
      consoleFormatter.format(createRecord({ level: 'alert' }))
    ).toMatchSnapshot()
    expect(
      consoleFormatter.format(
        createRecord({
          level: 'emergency'
        })
      )
    ).toMatchSnapshot()
  })

  it('should respect the provided color support', () => {
    const decoratedConsoleFormatter = new ConsoleFormatter({ decorated: true })
    expect(
      decoratedConsoleFormatter.format(record('debug', 'Test ConsoleFormatter'))
    ).toMatchSnapshot()

    const undecoratedConsoleFormatter = new ConsoleFormatter({
      decorated: true
    })
    expect(
      undecoratedConsoleFormatter.format(
        record('debug', 'Test ConsoleFormatter')
      )
    ).toMatchSnapshot()
  })
})
