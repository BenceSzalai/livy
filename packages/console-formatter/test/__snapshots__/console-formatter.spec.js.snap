// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@livy/console-formatter should format records correctly 1`] = `
"[2m2015-08-15 14:30:00 [22m[34m[2mDEBUG [22m[39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should format records correctly 2`] = `
"[2m2015-08-15 14:30:00 [22m[34mINFO [39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should format records correctly 3`] = `
"[2m2015-08-15 14:30:00 [22m[34mNOTICE [39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should format records correctly 4`] = `
"[2m2015-08-15 14:30:00 [22m[33mWARNING [39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should format records correctly 5`] = `
"[2m2015-08-15 14:30:00 [22m[31mERROR [39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should format records correctly 6`] = `
"[2m2015-08-15 14:30:00 [22m[31mCRITICAL [39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should format records correctly 7`] = `
"[2m2015-08-15 14:30:00 [22m[31mALERT [39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should format records correctly 8`] = `
"[2m2015-08-15 14:30:00 [22m[31mEMERGENCY [39mTest ConsoleFormatter
  [2mContext:[22m
  [33mfoo:[39m [36m1[39m
  
  [2mExtra:[22m
  [33mbar:[39m [36m2[39m
  "
`;

exports[`@livy/console-formatter should respect the provided color support 1`] = `"[2m2015-08-15 14:30:00 [22m[34m[2mDEBUG [22m[39mTest ConsoleFormatter"`;

exports[`@livy/console-formatter should respect the provided color support 2`] = `"[2m2015-08-15 14:30:00 [22m[34m[2mDEBUG [22m[39mTest ConsoleFormatter"`;
